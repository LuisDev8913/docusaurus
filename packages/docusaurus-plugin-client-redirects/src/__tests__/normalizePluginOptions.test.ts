/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import normalizePluginOptions, {
  DefaultPluginOptions,
} from '../normalizePluginOptions';
import type {CreateRedirectsFnOption} from '@docusaurus/plugin-client-redirects';

describe('normalizePluginOptions', () => {
  it('returns default options for undefined user options', () => {
    expect(normalizePluginOptions()).toEqual(DefaultPluginOptions);
  });

  it('returns default options for empty user options', () => {
    expect(normalizePluginOptions()).toEqual(DefaultPluginOptions);
  });

  it('overrides one default options with valid user options', () => {
    expect(
      normalizePluginOptions({
        toExtensions: ['html'],
      }),
    ).toEqual({...DefaultPluginOptions, toExtensions: ['html']});
  });

  it('overrides all default options with valid user options', () => {
    const createRedirects: CreateRedirectsFnOption = (_routePath: string) => [];
    expect(
      normalizePluginOptions({
        fromExtensions: ['exe', 'zip'],
        toExtensions: ['html'],
        createRedirects,
        redirects: [{from: '/x', to: '/y'}],
      }),
    ).toEqual({
      id: 'default',
      fromExtensions: ['exe', 'zip'],
      toExtensions: ['html'],
      createRedirects,
      redirects: [{from: '/x', to: '/y'}],
    });
  });

  it('rejects bad fromExtensions user inputs', () => {
    expect(() =>
      normalizePluginOptions({
        fromExtensions: [null, undefined, 123, true] as unknown as string[],
      }),
    ).toThrowErrorMatchingSnapshot();
  });

  it('rejects bad toExtensions user inputs', () => {
    expect(() =>
      normalizePluginOptions({
        toExtensions: [null, undefined, 123, true] as unknown as string[],
      }),
    ).toThrowErrorMatchingSnapshot();
  });

  it('rejects bad createRedirects user inputs', () => {
    expect(() =>
      normalizePluginOptions({
        createRedirects: ['bad', 'value'] as unknown as CreateRedirectsFnOption,
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});
