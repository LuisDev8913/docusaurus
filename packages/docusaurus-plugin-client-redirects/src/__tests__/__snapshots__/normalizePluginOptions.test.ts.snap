// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalizePluginOptions rejects bad createRedirects user inputs 1`] = `
"Invalid @docusaurus/plugin-client-redirects options: \\"createRedirects\\" must be of type function
  {
  \\"createRedirects\\": [
    \\"bad\\",
    \\"value\\"
  ]
}"
`;

exports[`normalizePluginOptions rejects bad fromExtensions user inputs 1`] = `
"Invalid @docusaurus/plugin-client-redirects options: \\"fromExtensions[0]\\" contains an invalid value
  {
  \\"fromExtensions\\": [
    null,
    null,
    123,
    true
  ]
}"
`;

exports[`normalizePluginOptions rejects bad toExtensions user inputs 1`] = `
"Invalid @docusaurus/plugin-client-redirects options: \\"toExtensions[0]\\" contains an invalid value
  {
  \\"toExtensions\\": [
    null,
    null,
    123,
    true
  ]
}"
`;
